doc/doc-support/python-interpreter-table/collect-data/nix/toComparable.nix

```diff
@@ -0,0 +1,32 @@
+/* Convert a "<name> <version>" string to a comparable list.
+
+The purpose of this function is to allow for
+natural sorting of strings representing versioned names,
+where the version are numbers separated by dots.
+
+Examples:
+
+- `toComparable "CPython"` returns `[ "CPython" 0 ]`
+- `toComparable "CPython 3"` returns `[ "CPython" 3 ]`
+- `toComparable "CPython 3.0"` returns `[ "CPython" 3 0 ]`
+- `toComparable "CPython 3.9"` returns `[ "CPython" 3 9 ]`
+- `toComparable "CPython 3.10"` returns `[ "CPython" 3 10 ]`
+*/
+{ lib }:
+str:
+let
+  inherit (lib.lists) head length map tail;
+  inherit (lib.strings) splitString toInt;
+  inherit (lib.versions) splitVersion;
```

Comment:
```suggestion
  inherit (lib) head length tail splitString toInt splitVersion;
```

Reply:
Doesn't that defy the whole purpose of namespacing `lib`?

Reply:
The namespacing is internal to lib, we shouldn't use it outside if there is no good reason.

Reply:
No the namespacing into sublibraries isn't internal (if it was, you couldn't access it from the outside!). Most functions are accessible from the top-level namespace though, so it's not a problem to mostly ignore the namespacing unless you need functions that aren't exposed from the top-level, or if you want to make the code a bit clearer.

This might change in the future with a `lib` namespace refactoring.